### PR TITLE
feat(AutoSuggestField Menu): Fixes debounce issue and menu

### PR DESCRIPTION
**Pull request by yasabere**
affects: @rentpath/react-ui-core

[Card](https://rentpath.leankit.com/card/634890555)

- Fixes debounce issue
- Added ability to include non-clickable options in suggestions

### CODE REVIEW

**Comment by tadjohnston** [comment]
Where is the card for this and if you add something it has to have unit tests with it.

File: packages/react-ui-core/src/Menu/Menu.js [hunk 1]
```diff
@@ -100,21 +108,49 @@ export default class Menu extends PureComponent {
...
+    this.setState({
+      highlightIndex: this.state.indexedOptions[index].index,
+      indexedOptionIndex: index,
+    }, () => {
       if (onItemMouseOver) onItemMouseOver(this.value)
     })
   }
 
+  @autobind
+  generateIndexedOption() {
```
Comment (nray621) [inline packages/react-ui-core/src/Menu/Menu.js]: nitpick: Can we call this `generateIndexedOptions` with an s?

**Comment by yasabere** [comment]
Added tests

**Comment by bethstevic** [comment]
Looks good to me 👍

**Comment by tadjohnston** [comment]
I still don't know what ticket this PR is resolving.  Can we link one with an explanation of the debounce issue?

**Pull request merged**